src/main.test.ts

```diff
@@ -1,88 +1,491 @@
+    );
+
+    result.savedImages.forEach((savedImage: SavedImage) => {
+      const matchedSnapshot = snapshots.find(snapshot =>
+        snapshot.includes(savedImage.name),
+      );
+
+      expect(
+        doFilesLookSame(
+          savedImage.path,
+          path.join(SNAPSHOT_PATH, testSuite, matchedSnapshot as string),
+        ),
+      ).toBeTruthy();
+    });
+  };
+
+  describe('using a local input', () => {
+    test('in png format', async () => {
+      const testSuite = 'input-png';
+      const result = await generateTempImages(
+        {
+          scrape: false,
+          log: false,
+          background: 'coral',
+        },
+        './static/logo.png',
+        `./temp/local/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
+
+    test('in svg format', async () => {
+      const testSuite = 'input-svg';
+      const result = await generateTempImages(
+        {
+          scrape: false,
+          log: false,
+          background: 'coral',
+        },
+        './static/logo.svg',
+        `./temp/local/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
+
+    test('in html format', async () => {
+      const testSuite = 'input-html';
+      const result = await generateTempImages(
+        {
+          scrape: false,
+          log: false,
+        },
+        './static/logo.html',
+        `./temp/local/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
+
+    test('with a transparency', async () => {
+      const testSuite = 'output-transparent';
+      const result = await generateTempImages(
+        {
+          scrape: false,
+          log: false,
+          background: 'transparent',
+          opaque: false,
+        },
+        './static/logo.svg',
+        `./temp/local/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
+
+    test('with JPG format as output', async () => {
+      const testSuite = 'output-jpeg';
+      const result = await generateTempImages(
+        {
+          scrape: false,
+          log: false,
+          background: 'coral',
+          type: 'jpeg',
+          quality: 80,
+        },
+        './static/logo.svg',
+        `./temp/local/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
   });
 
-  expect(result).toMatchSnapshot();
+  describe('using a remote input', () => {
+    test('in png format', async () => {
+      const testSuite = 'input-png';
+      const result = await generateTempImages(
+        {
+          scrape: false,
+          log: false,
+          background: 'coral',
+        },
+        'https://raw.githubusercontent.com/onderceylan/pwa-asset-generator/HEAD/static/logo.png',
+        `./temp/remote/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
+
+    test.skip('in svg format', async () => {
```

Comment:
Skipped tests have a dependency on static files hosted on master, I will enable them once this PR is merged.